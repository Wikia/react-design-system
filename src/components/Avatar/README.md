Defaults:
```js
<Avatar />
```

Custom image and classname:
```js
<Avatar src="https://bit.ly/2ALKN1o" className="highligted-avatar"/>
```

Custom image, classname, and link:
```js
<Avatar src="https://bit.ly/2ALKN1o" className="highligted-avatar" link="http://google.com"/>
```
