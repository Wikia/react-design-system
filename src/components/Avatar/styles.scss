.avatar {
	display: block;
	border: solid 2px #1a1a1a;
	border-radius: 50%;
	height: 35px;
	overflow: hidden;
	width: 35px;

	&:hover {
		border-color: #00b7e0;
	}

	img {
		height: 100%;
		object-fit: cover;
		width: 100%;
	}
}
