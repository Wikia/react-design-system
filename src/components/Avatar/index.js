import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const Avatar = (props) => {
  if (props.link) {
    return (
      <a className={`avatar ${props.className}`} href={props.link}>
        <img src={props.src} alt={props.alt} />
      </a>
    );
  }

  return (
    <div className={`avatar ${props.className}`}>
      <img src={props.src} alt={props.alt} />
    </div>
  );
};

Avatar.propTypes = {
  /**
   * Additional classname to append to classlist
   */
  className: PropTypes.string,
  /**
   * source of an image file for the avatar image
   */
  src: PropTypes.string,
  /**
   * Alt text filled in the source image
   */
  alt: PropTypes.string,
  /**
   * If provided will be wrapped in anchor tag
   */
  link: PropTypes.string,
};

Avatar.defaultProps = {
  className: '',
  link: '',
  alt: 'avatar',
  src: 'https://vignette.wikia-dev.us/messaging/images/1/19/Avatar.jpg/revision/latest/scale-to-width-down/50',
};

export default Avatar;
