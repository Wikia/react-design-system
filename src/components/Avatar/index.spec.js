import React from 'react';
import renderer from 'react-test-renderer';

import Avatar from './index';

test('Avatar renders correctly with no props', () => {
  const component = renderer.create(
    <Avatar />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});


test('Avatar renders correctly with extra classsname', () => {
  const component = renderer.create(
    <Avatar className="special-class" />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});


test('Avatar renders correctly with extra classsname and source', () => {
  const component = renderer.create(
    <Avatar
      className="special-class"
      src="https://vignette.wikia.nocookie.net/af9f47aa-d4ec-420d-a6b9-6d38644e5644/scale-to-width-down/20"
    />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Avatar renders correctly with extra classsname, source, and link', () => {
  const component = renderer.create(
    <Avatar
      className="special-class"
      src="https://vignette.wikia.nocookie.net/af9f47aa-d4ec-420d-a6b9-6d38644e5644/scale-to-width-down/20"
      link="http://google.com"
    />,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Avatar renders correctly with children', () => {
  const component = renderer.create(
    <Avatar> C </Avatar>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test('Avatar renders correctly with alt border', () => {
  const component = renderer.create(
    <Avatar borderColor="green"> C </Avatar>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
