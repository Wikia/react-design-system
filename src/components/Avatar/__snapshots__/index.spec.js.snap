// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders correctly with extra classsname 1`] = `
<div
  className="avatar special-class"
>
  <img
    alt="avatar"
    src="https://vignette.wikia-dev.us/messaging/images/1/19/Avatar.jpg/revision/latest/scale-to-width-down/50"
  />
</div>
`;

exports[`Avatar renders correctly with extra classsname and source 1`] = `
<div
  className="avatar special-class"
>
  <img
    alt="avatar"
    src="https://vignette.wikia.nocookie.net/af9f47aa-d4ec-420d-a6b9-6d38644e5644/scale-to-width-down/20"
  />
</div>
`;

exports[`Avatar renders correctly with extra classsname, source, and link 1`] = `
<a
  className="avatar special-class"
  href="http://google.com"
>
  <img
    alt="avatar"
    src="https://vignette.wikia.nocookie.net/af9f47aa-d4ec-420d-a6b9-6d38644e5644/scale-to-width-down/20"
  />
</a>
`;

exports[`Avatar renders correctly with no props 1`] = `
<div
  className="avatar "
>
  <img
    alt="avatar"
    src="https://vignette.wikia-dev.us/messaging/images/1/19/Avatar.jpg/revision/latest/scale-to-width-down/50"
  />
</div>
`;
